build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/RestrictedBuildApiService.java

```diff
@@ -67,6 +67,7 @@ private static ListMultimap<Class<?>, String> createLegacyRestTestBasePluginUsag
         map.put(LegacyRestTestBasePlugin.class, ":qa:remote-clusters");
         map.put(LegacyRestTestBasePlugin.class, ":qa:repository-multi-version");
         map.put(LegacyRestTestBasePlugin.class, ":qa:rolling-upgrade");
```

Comment:
we should be able to remove `:qa:rolling-upgrade` from this list if the project doesn't use any legacy plugins anymore